src/main/java/echoserver/EchoServerService.java

```diff
@@ -0,0 +1,38 @@
+package echoserver;
+
+import java.io.IOException;
+import java.util.concurrent.Executor;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
+public class EchoServerService {
+
+
+    public EchoServerService(ConnectionHub hostServer, Logger logger, ThreadPoolExecutor connectionThreadSpool) {
```

Comment:
To follow up on an earlier comment. While I think it is important to use an existing implementation of the thread pool executor, I wonder if we want to *depend* on the concrete type here, or to depend on the interface. I think that interface would be the `ExecutorService` mayhaps?

https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ExecutorService.html